### ISSUE TITLE
Microsecond precision

### ISSUE THREAD

**Issue by seansawyer** [erlsci/iso8601#7]
This will need to be a major version bump, since the time() type will have to change to accomodate microseconds.

**Comment by lordmetroid** [comment]
I would love to see some microseconds included someday.

**Comment by oubiwann** [comment]
PR #16 adds support for this using ``parse_exact`` (in order to not break the API).

It currently needs to be rebased and then have some unit tests added ...

**Comment by aseigo** [comment]
PR #31 was merged that was an improved version of PR #16, so this could now be closed?

**Comment by KrzysiekJ** [comment]
Using floats is scary when speaking about high precision.

```
1> 1 - 0.9.
0.09999999999999998
2> 1.1 - 1.
0.10000000000000009
```

Does the current implementation of `parse_exact/1` guarantee to be safe in regard to rounding? For example, if seconds are written as `01.2345000000`, then if I extract miliseconds as `round((SecsFloat - trunc(SecsFloat)) * 1000)`, will it always yield 235?

**Comment by oubiwann** [comment]
Okay, in preparation for 1.4, I did some digging and here's what I found:

parse_exact builds the fraction with list_to_float("0.234500") — it parses the literal string, it does not do the kind of subtraction that produces the scary 1 - 0.9 = 0.0999…8. That classic example is arithmetic-accumulation error; parse_exact doesn't accumulate. Checking the IEEE-754 reality (language-independent — iiuc, Erlang floats are the same doubles):

```
0.1234565 * 1000 = 123.45649999999999 -> rounds to 123 (fine for ms)
0.234500 * 1e6 = 234500.0 -> exact
59.999999 * 1e6 = 59999999.0 -> exact
```

For inputs at genuine microsecond precision (≤6 fractional digits), the best I can tell is that the residual representation error is on the order of 1e-7 or smaller — far below the 0.5 rounding threshold — so round((Frac) * 1e6) recovers the intended integer reliably. It seems like the only place it might bite us is exact half-unit boundaries, and those require a 7th decimal digit of 5 (i.e. sub-microsecond input), which is below the precision you're claiming anyway.

So, long story short: I'm cool with closing this out :-D
